src/TCommandLine.cpp

```diff
@@ -996,15 +996,25 @@ void TCommandLine::handleTabCompletion(bool direction)
     buffer.replace(QChar::LineFeed, QChar::Space);
 
     QStringList wordList = buffer.split(QRegularExpression(qsl(R"(\b)"), QRegularExpression::UseUnicodePropertiesOption), Qt::SkipEmptyParts);
+    wordList.append(commandLineSuggestions.values()); // hindsight 20/20 I do not need to split this to a separate table, a check to not append buffer to this table and only append suggested list does same thing for far less overhead. 
+    QStringList blacklist = tabCompleteBlacklist.values(); 
+    QStringList toDelete;
 
-    wordList.append(commandLineSuggestions.values());
+    foreach(QString wstr, wordList) {
+        if (blacklist.contains(wstr, Qt::CaseInsensitive)) {
+            toDelete += wstr;
+        }
+    }
+    foreach(QString dstr, toDelete) {
+        wordList.removeAll(dstr);
+    }
 
     if (direction) {
         mTabCompletionCount++;
     } else {
         mTabCompletionCount--;
     }
-    if (!wordList.empty()) {
+    if (!wordList.empty()) { 
```

Comment:
Bogus extra whitespace.
```suggestion
    if (!wordList.empty()) {
```